aiida_common_workflows/workflows/relax/siesta/generator.py

```diff
@@ -0,0 +1,255 @@
+# -*- coding: utf-8 -*-
+"""
+Inputs generator for the SiestaRelaxWorkChain
+"""
+import os
+import yaml
+from aiida.common import exceptions
+from aiida.orm.groups import Group
+from aiida_common_workflows.workflows.relax.generator import RelaxInputsGenerator, RelaxType
+from aiida_common_workflows.workflows.relax.siesta.workchain import SiestaRelaxWorkChain
+
+__all__ = ('SiestaRelaxInputsGenerator',)
+
+
+class SiestaRelaxInputsGenerator(RelaxInputsGenerator):
+    """Generator of inputs for the SiestaRelaxWorkChain"""
+
+    _default_protocol = 'moderate'
+
+    _filepath = os.path.join(os.path.dirname(__file__), 'protocols_registry.yaml')
+
+    with open(_filepath) as _thefile:
+        _protocols = yaml.full_load(_thefile)
```

Comment:
would it be better to have this in a method. Always a bit hesitant with opening and closing files in class definitions outside of a method